LogcatCoreLib/src/main/java/info/hannes/logcat/LogfileFragment.kt

```diff
@@ -10,30 +10,21 @@ import info.hannes.timber.FileLoggingTree
 import info.hannes.timber.fileLoggingTree
 import java.io.File
 
-class LogfileFragment : LogBaseFragment(), Observer<String> {
+class LogfileFragment : LogBaseFragment(), Observer<Event<String>> {
 
     private var sourceFileName: String? = null
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
-        sourceFileName = fileLoggingTree()?.getFileName()
-        return super.onCreateView(inflater, container, savedInstanceState)
-    }
-
-    override fun onStart() {
-        super.onStart()
-        FileLoggingTree.lastLogEntry.observeForever(this)
-    }
-
-    override fun onStop() {
-        FileLoggingTree.lastLogEntry.removeObserver(this)
-        super.onStop()
+        val view = super.onCreateView(inflater, container, savedInstanceState)
+        FileLoggingTree.lastLogEntry.observe(viewLifecycleOwner, this)
```

Comment:
Hmm, now you observe it always. I guess because of the duplicate-last-line-issue. I'm not 100 % convinced

Reply:
But observation is tied to lifecycle, so it's not always observed, maybe this article sheds more light, https://medium.com/@cs.ibrahimyilmaz/viewlifecycleowner-vs-this-a8259800367b